main.go

```diff
@@ -6,45 +6,68 @@ package main
 
 import (
 	"fmt"
-	"log"
+	"os"
 
 	"github.com/docopt/docopt-go"
 	"github.com/taskcluster/taskcluster-worker/engines/extpoints"
+	"github.com/taskcluster/taskcluster-worker/runtime"
 )
 
 const version = "taskcluster-worker 0.0.1"
 const usage = `
-Usage: taskcluster-worker [options]
-Runs a worker with the given options.
-Options:
-  -V, --version            Display the version of go-import-subtree and exit.
-  -h, --help               Print this help information.
-  -e, --engine <engine>    Execution engine to run tasks in
+TaskCluster worker
+This worker is meant to be used with the taskcluster platform for the execution and
+resolution of tasks.
+
+  Usage:
+    taskcluster-worker --help
+    taskcluster-worker --version
+    taskcluster-worker --engine <engine>
+    taskcluster-worker --engine <engine> --logging <level>
+
+  Options:
+    --help  						Show this help screen.
+    --version  						Display the version of go-import-subtree and exit.
+    -e --engine <engine>  			Engine to use for task execution sandboxes.
+    -l --logging-level <level>  	Set logging at <level>.
 `
 
 func main() {
 	args, err := docopt.Parse(usage, nil, true, version, false, true)
 	if err != nil {
-		log.Fatalf("Error parsing arguments. %v", err)
+		fmt.Fprintf(os.Stderr, "Error parsing arguments. %v", err)
+		os.Exit(1)
 	}
 
-	e := args["--engine"]
-	if e == nil {
-		panic("Must supply engine type")
+	var level string
+	if l := args["--logging-level"]; l != nil {
+		level = l.(string)
+	}
+	logger, err := runtime.CreateLogger(level)
+	if err != nil {
+		os.Stderr.WriteString(err.Error())
+		os.Exit(1)
```

Comment:
We do not need a stack trace, nor do we need to run deferred functions, so panic does not seem like the best suited option here.  This is an error we can't recover from, and should exit immediately.  It also allows us to define exit codes for things we know went wrong.

Panic seems more appropriate when it happens lower down in the execution of the worker (like if we had a plugin hit some unrecoverable state and we need to unwind the stack all the way up) and complete any deferred functions for cleanup.

For my understanding of what is idiomatic in Go, and in the case of using panic here, what is elegant about using panic vs this method of exiting?